protocols/request-response/src/cbor.rs

```diff
@@ -130,24 +133,43 @@ mod codec {
         where
             T: AsyncWrite + Unpin + Send,
         {
-            let data: Vec<u8> = serde_cbor::to_vec(&resp).map_err(into_io_error).unwrap();
+            let data: Vec<u8> =
+                cbor4ii::serde::to_vec(Vec::new(), &resp).map_err(IntoError::into_io_error)?;
 
             io.write_all(data.as_ref()).await?;
 
             Ok(())
         }
     }
 
-    fn into_io_error(err: serde_cbor::Error) -> io::Error {
-        if err.is_syntax() || err.is_data() {
-            return io::Error::new(io::ErrorKind::InvalidData, err);
-        }
+    trait IntoError {
+        fn into_io_error(self) -> io::Error;
+    }
 
-        if err.is_eof() {
-            return io::Error::new(io::ErrorKind::UnexpectedEof, err);
+    impl IntoError for cbor4ii::serde::DecodeError<Infallible> {
+        fn into_io_error(self) -> io::Error {
+            match self {
+                cbor4ii::serde::DecodeError::Core(e) => match e {
+                    DecodeError::Read(e) => io::Error::new(io::ErrorKind::Other, e),
```

Comment:
These `match` blocks can be collapsed:

```suggestion
            match self {
                cbor4ii::serde::DecodeError::Core(DecodeError::Read(e)) => io::Error::new(io::ErrorKind::Other, e),
                // etc
```

Reply:
@thomaseizinger I didn't collapse these because I want to use the match on the inner error (for eof/unsupported) to ascribe the right io::Error, and still use the library's display fun on the parent errors. I rather not create the error from the destructure.

Reply:
for `Read` we don't need the inner match, but it's really b/c Eof/Unexpected have structure's underneath.

Reply:
Nvm, the lib does indeed provide the display trait on the parent wrapper error. +1.